File: Leetcode/Simplify Path.cc
class Solution {
public:
    string simplifyPath(string path) {
        vector<string> path_vec;
        string now;
        path += '/';
        for (int i = 0; path[i]; /*PASS*/) {
            if (path[i] == '/') {
                if (now == "."  || now == "") {
                    // PASS
                } else if (now == "..") {
                    if (!path_vec.empty()) {
                        path_vec.pop_back();
                    }
                } else {
                   path_vec.push_back(now); 
                }
                now = "";
                while (path[i] == '/') {
                    i++;
                }
            } else {
                now += path[i];
                i++;
            }
        }
        string res;
        for(int i = 0; i < path_vec.size(); i++) {
            res += '/';
            res += path_vec[i];
        }
        return res == ""? "/" : res;
    }
};

File: LintCode/python/Simplify Path.py
class Solution:
    # @param {string} path the original path
    # @return {string} the simplified path
    def simplifyPath(self, path):
        path = filter(lambda x: x,
                path.split('/'))
                
        res = []
        for item in path:
            if item == '.':
                continue
            if item == '..':
                if res:
                    res.pop()
                continue
            res.append(item)
            
        return '/' + '/'.join(res)
